pkg/identityserver/end_device_registry.go

```diff
@@ -228,6 +228,37 @@ func (is *IdentityServer) updateEndDevice(ctx context.Context, req *ttnpb.Update
 	return dev, nil
 }
 
+func (is *IdentityServer) batchUpdateEndDevice(ctx context.Context, req *ttnpb.BatchUpdateEndDeviceRequest) (*ttnpb.EndDevices, error) {
+	if err := clusterauth.Authorized(ctx); err != nil {
+		return nil, err
+	}
+	req.FieldMask = cleanFieldMaskPaths([]string{"activated_at", "locations", "last_seen"}, req.FieldMask, nil, getPaths)
```

Comment:
I would just allow only `last_seen_at` for now, also would simplify the logic below.